frontend/src/pages/register/form.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { useContext, useState } from 'react'
+import { Routes } from '../../enums/routes'
+import { Form } from '../../components'
+import { Title, AccentText, Link } from '../../components/text/text-styles'
+import { AuthContext } from '../../providers/authProvider'
+
+const RegisterForm = () => {
+  const [email, setEmail] = useState<string>('')
+  const [password, setPassword] = useState<string>('')
+  const [confirmPassword, setConfirmPassword] = useState<string>('')
+  const [firstName, setFirstName] = useState<string>('')
+  const [lastName, setLastName] = useState<string>('')
+  const [passwordVisible, setPasswordVisible] = useState<boolean>(false)
+  const [confirmPasswordVisible, setConfirmPasswordVisible] =
+    useState<boolean>(false)
+  const [error, setError] = useState<string>()
+
+  const { register, loading } = useContext(AuthContext)
+  const data = { email, password, firstName, lastName }
+
+  const emailChangeHandler = (event: React.FormEvent<HTMLInputElement>) => {
+    setEmail(event.currentTarget.value)
+  }
+
+  const firstNameChangeHandler = (event: React.FormEvent<HTMLInputElement>) => {
```

Comment:
Seems like we have lots of duplication when it comes to these handlers. There should be a better way to handle form fields

Reply:
@sanja-htec  fixed this on her branch